horovod/common/operations.cc

```diff
@@ -1342,7 +1395,11 @@ void BackgroundThreadLoop(HorovodGlobalState& state) {
   MPI_Comm_free(&state.local_comm);
   MPI_Comm_free(&state.cross_comm);
 
-  MPI_Finalize();
+#if HAVE_DDL
+  ddl_finalize();
+#else
+   MPI_Finalize();
```

Comment:
Can you add a comment that `ddl_finalize()` performs MPI finalization as well?

Nit: remove one extra space before `MPI`